### PR TITLE
feat: Retrieve Cohort Replay

### PR DESCRIPTION
**Pull request by Warren-Pitterson**
## Description

Retrieve Cohort Distribution by Request Id

## Context

Allows the user to get a previous request retrieved request by the Id

## Type of changes

- [ ] Refactoring (non-breaking change)
- [ x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x ] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [x ] I have followed the code style of the project
- [x ] I have added tests to cover my changes
- [ ] I have updated the documentation accordingly
- [ ] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [x ] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

File: application/CohortManager/src/Functions/CohortDistributionServices/RetrieveCohortReplay/RetrieveCohortReplay.cs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    private readonly HttpRequestHelper _httpRequestHelper;
+    public RetrieveCohortReplay(ILogger<RetrieveCohortReplay> logger, ICreateCohortDistributionData createCohortDistributionData, ICreateResponse createResponse, IExceptionHandler exceptionHandler, HttpRequestHelper httpRequestHelper)
+    {
+        _logger = logger;
+        _createCohortDistributionData = createCohortDistributionData;
+        _createResponse = createResponse;
+        _exceptionHandler = exceptionHandler;
+        _httpRequestHelper = httpRequestHelper;
+    }
+
```
Comment (will-larkin-nhs) [inline application/CohortManager/src/Functions/CohortDistributionServices/RetrieveCohortReplay/RetrieveCohortReplay.cs:27]: XML Docs here pls, example in the transform service

**Review by MWClayson-NHS — COMMENTED** [review]
Looks good! One small comment and can you sort the pipeline issue?

File: application/CohortManager/src/Functions/CohortDistributionServices/RetrieveCohortReplay/RetrieveCohortReplay.cs (hunk 1 above)
Comment (Warren-Pitterson) [reply application/CohortManager/src/Functions/CohortDistributionServices/RetrieveCohortReplay/RetrieveCohortReplay.cs:41]: added

File: application/CohortManager/src/Functions/CohortDistributionServices/RetrieveCohortReplay/Program.cs [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    .ConfigureServices(services =>
+    {
+        DbProviderFactories.RegisterFactory("System.Data.SqlClient", SqlClientFactory.Instance);
+        services.AddTransient<IDbConnection>(provider =>
+        {
+            var providerFactory = DbProviderFactories.GetFactory("System.Data.SqlClient");
+            var conn = providerFactory.CreateConnection();
+            return conn;
+        });
+        services.AddSingleton<ICreateCohortDistributionData, CreateCohortDistributionData>();
```
[the remark this answers is not in the document]
Comment (Warren-Pitterson) [reply application/CohortManager/src/Functions/CohortDistributionServices/RetrieveCohortReplay/Program.cs]: changed

**Review by MWClayson-NHS — DISMISSED** [review]
Good work, Approved!

**Review by MWClayson-NHS — DISMISSED** [review]

**Review by will-larkin-nhs — APPROVED** [review]

**Pull request merged**